packages/dapp/src/reducers/listings.ts

```diff
@@ -67,6 +68,18 @@ export function histories(
   }
 }
 
+export function listingHistorySubscriptions(
+  state: Map<string, Subscription> = Map<string, Subscription>(),
+  action: AnyAction,
+): Map<string, Subscription> {
+  switch (action.type) {
+    case listingActions.ADD_HISTORY_SUBSCRIPTION:
+      return state.set(action.data.address, action.data.subscription);
```

Comment:
theres no check to make sure you don't overwrite an already existing subscription.  That could cause memory leaks.

Reply:
I check in the action creator before dispatching. What's the right thing to do here in the reducer? I thought I can't really do anything like unsubscribe from an existing subscription in the reducer since reducers can't have side effects, right?

Reply:
Checking in the action is fine, i missed that.  I would have only set it if it was not already set but they're the same.